src/triage/component/postmodeling/error_analysis.py

```diff
@@ -0,0 +1,169 @@
+                test_matrix_thresh['above_thresh']
+    elif param_type == 'rank_pct':
+        # Calculate residuals/errors
+        test_matrix_thresh = test_matrix.sort_values(['rank_pct'], ascending=True)
+        test_matrix_thresh['above_thresh'] = \
+                np.where(test_matrix_thresh['rank_pct'] <= param, 1, 0)
+        test_matrix_thresh['error'] = test_matrix_thresh['label_value'] - \
+                test_matrix_thresh['above_thresh']
+    else:
+        raise AttributeError('''Error! You have to define a parameter type to
+                             set up a threshold
+                             ''')
+
+    # Define labels using the errors
+    dict_errors = {'FP': (test_matrix_thresh['label_value'] == 0) &
+                          (test_matrix_thresh['above_thresh'] == 1),
+                   'FN': (test_matrix_thresh['label_value'] == 1) &
+                          (test_matrix_thresh['above_thresh'] == 0),
+                   'TP':  (test_matrix_thresh['label_value'] == 1) &
+                          (test_matrix_thresh['above_thresh'] == 1),
+                   'TN':  (test_matrix_thresh['label_value'] == 0) &
+                          (test_matrix_thresh['above_thresh'] == 0)
+                  }
+    test_matrix_thresh['class_error'] = np.select(condlist=dict_errors.values(),
+                                                 choicelist=dict_errors.keys(),
+                                                 default=None)
+
+    # Split data frame to explore FPR/FNR against TP and TN
+    test_matrix_thresh_0 = \
+    test_matrix_thresh[test_matrix_thresh['label_value'] == 0]
+    test_matrix_thresh_1 = \
+    test_matrix_thresh[test_matrix_thresh['label_value'] == 1]
+    test_matrix_predicted_1 = \
+    test_matrix_thresh[test_matrix_thresh['above_thresh'] == 1]
+
+    dict_error_class = {'FPvsAll': (test_matrix_thresh['class_error'] == 'FP'),
+                        'FNvsAll': (test_matrix_thresh['class_error'] == 'FN'),
+                        'FNvsTP': (test_matrix_thresh_1['class_error'] == 'FN'),
+                        'FPvsTN': (test_matrix_thresh_0['class_error'] == 'FP'),
+                        'FPvsTP': (test_matrix_predicted_1['class_error'] == 'FP')}
+
+    # Create label iterator
+    Y = [(np.where(condition, 1, -1), label) for label, condition in \
+         dict_error_class.items()]
+
+    # Define feature space to model: get the list of feature names
+    storage = ProjectStorage(path)
+    matrix_storage = MatrixStorageEngine(storage).get_store(self.pred_matrix_uuid)
+    feature_columns = matrix_storage.columns()
+
+    # Build error feature matrix
+    matrices = [test_matrix_thresh,
+                test_matrix_thresh,
+                test_matrix_thresh_1,
+                test_matrix_thresh_0,
+                test_matrix_predicted_1]
+    X = [matrix[feature_columns] for matrix in matrices]
+
+    return zip(Y, X)
+
+def _error_modeler(self,
```

Comment:
I notice that both `_error_labeler` and `_error_modeler` are referred to in other functions' `__doc__` – should they really be "hidden" from the interface, by that leading underscore?